### DISCUSSION TITLE
How to add WORKSPACE Maven Installation

### DISCUSSION THREAD

**Discussion by tivervac** [eclipse-m2e/m2e-core#1335] [Q&A]
When I explore my installations, I see that it mentions I have no `WORKSPACE` installation available

![image](https://user-images.githubusercontent.com/3389524/227902614-e7baac96-6345-49ab-9ed7-d4e2caa9ae3e.png)

This entry is not editable. If I try to add a new installation, I'm only given the option for External installations, which doesn't seem to allow relative path or the [external tool variables](https://help.eclipse.org/latest/index.jsp?topic=%2forg.eclipse.platform.doc.user%2fconcepts%2fconcepts-exttools.htm).

![image](https://user-images.githubusercontent.com/3389524/227903479-6c3d9f29-dd21-4b24-9c04-debe6b4fb947.png)

How can I add a workspace maven installation? Is it supposed to be provided by a plugin?

**Comment by fbricon** [comment]
This is used when the Apache Maven project itself is imported in the workspace, and can then be used to execute Maven goals from eclipse. Basically, it's useful to Maven developers/ power users
Also see https://kthoms.wordpress.com/2019/11/27/eclipse-m2e-how-to-use-a-workspace-maven-installation/

**Reply by tivervac** [reply]
The blog post explains the `WORKSPACE` perfectly, thanks. Can a relative path or Eclipse variables somehow be used for the External maven installations? Or should it always be an absolute path? If so, is there a technical reason why M2E implemented this restriction?

**Reply by laeubi** [reply]
I don't see how a relative path would help here anything as one usually reference an **installation** that means it is part of your local computer, technically I think m2e can support **variables** in the path that then might point to a workspace location.

**Reply by tivervac** [reply]
I might be misunderstanding what a `Maven Installation` is to M2E. This discussion, to me, is still related to my [previous one](https://github.com/eclipse-m2e/m2e-core/discussions/1273#discussioncomment-5124137). There you mentioned  I can `control the maven version used to launch external maven runs`. This is what I'm trying to do, but I'm trying to do so in general, not for just one launch configuration. The only place I found that supplies a generic maven location is the M2E Maven Installation preference page.

**Reply by tivervac** [reply]
@laeubi Any way to achieve what I'm doing above? Is there any way to change the default maven installation used for new maven launch profiles? If not, is there a way to generically define a maven installation for the maven wrapper in the repo such that I can point to that installation in our existing maven launch profiles?

**Reply by laeubi** [reply]
Many things are possible, what I can think of is that we use something similar like we did for JVMS and the enforcer plugin to select a different maven by default, but I'm not aware of a workaround right now to fully archive what you want, so you have the following options:

1. Workaround the problem somehow by using an absolute path (what might still point at you workspace) and modify the run configs manually for the first time, you can then store the run in your workspace.
2. [Start hacking at m2e](https://github.com/eclipse-m2e/m2e-core/blob/master/CONTRIBUTING.md) and provide a PR that adds the missing feature, or hope that it is important enough for someone else to do so in the future.
3. If its crucial to your business you can hire me to implement certain features for m2e (or other open source projects)
